src/chatd.cpp

```diff
@@ -2505,6 +2505,17 @@ uint64_t Chat::publicHandle()
     return mPh;
 }
 
+bool Chat::previewMode()
+{
+    return mPh.isValid();
+}
+
+void Chat::rejoin()
+{
+    clearHistory();
+    join();
```

Comment:
Do we really need to send a new `JOIN` in this case? I though it would be enough with a `HIST`. If so, we should wrap the same code for `REJECT RANGE 1`:
```
void Chat::refreshHistory()
{
   clearHistory();
   requestHistoryFromServer(-chat.initialHistoryFetchCount);
}
```
And call it in that existing case and also in this new case (avoid the creation of `rejoin()` if so, and call `refreshHistory()` instead).